apps/heureka/src/components/filters/FilterSelect.jsx

```diff
@@ -5,52 +5,29 @@
 
 import React, { useState } from "react"
 import { Button, InputGroup, SelectOption, Select, Stack, SearchInput } from "@cloudoperators/juno-ui-components"
-import {
-  useFilterLabels,
-  useFilterLabelValues,
-  useFilterActions,
-  useActiveFilters,
-  useSearchTerm,
-} from "../../hooks/useAppStore"
+import { useFilterActions } from "../../hooks/useAppStore"
 import { humanizeString } from "../../lib/utils"
 
-const FilterSelect = ({ entityName, isLoading }) => {
+const FilterSelect = ({ entityName, isLoading, filterLabels, filterLabelValues, activeFilters, searchTerm }) => {
   const [filterLabel, setFilterLabel] = useState("")
   const [filterValue, setFilterValue] = useState("")
 
   const { addActiveFilter, clearActiveFilters, setSearchTerm } = useFilterActions()
 
-  const filterLabels = useFilterLabels(entityName)
-  const filterLabelValues = useFilterLabelValues(entityName)
-  const activeFilters = useActiveFilters(entityName)
-  const searchTerm = useSearchTerm()
-
   const handleFilterAdd = (value) => {
     if (filterLabel && (filterValue || value)) {
-      // Add the active filter to the store
       addActiveFilter(entityName, filterLabel, filterValue || value)
```

Comment:
This is NOT a change request it is a proposal :)
Did you think about to pass the addActiveFilter as callback to the component und make the component entity agnostic?

Reply:
Sure, this is a valuable optimization, and I implement it.